src/main/java/seedu/address/model/person/Frequency.java

```diff
@@ -0,0 +1,85 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a how often a client's policy is updated in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidFrequency(String)}
+ */
+public class Frequency {
+
+    enum Months {
+        JANUARY, FEBRUARY, MARCH, APRIL, MAY, JUNE, JULY, AUGUST, SEPTEMBER, OCTOBER, NOVEMBER, DECEMBER
+    }
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Frequency can only be 1, 3, 6 or 12 months";
+
+    public final String value;
+    public final Months[] months;
+
+    /**
+     * Constructs a {@code Frequency}.
+     * Hard coded months for each frequency.
+     * @param frequency
+     * @throws IllegalArgumentException
+     */
+    public Frequency(String frequency) throws IllegalArgumentException {
+        requireNonNull(frequency);
+        checkArgument(isValidFrequency(frequency), MESSAGE_CONSTRAINTS);
+        this.value = frequency;
+        switch (frequency) {
+        case "0":
+            this.months = new Months[]{};
+            break;
+        case "1":
+            this.months = new Months[]{Months.JANUARY, Months.FEBRUARY, Months.MARCH, Months.APRIL, Months.MAY,
+                Months.JUNE, Months.JULY, Months.AUGUST, Months.SEPTEMBER,
+                Months.OCTOBER, Months.NOVEMBER, Months.DECEMBER};
+            break;
+        case "3":
+            this.months = new Months[]{Months.JANUARY, Months.APRIL, Months.JULY, Months.OCTOBER};
+            break;
+        case "6":
+            this.months = new Months[]{Months.JANUARY, Months.JULY};
+            break;
+        case "12":
+            this.months = new Months[]{Months.JANUARY};
+            break;
```

Comment:
Not sure what this portion is used for. Can I take this to mean that e.g. should a client's policy have a renewal of 6 months, they are only allowed to pay on January and July, instead of possibly March and September?

Reply:
As per what we discussed we will take it as these are the only months in which payment can be processed